pydantic/types.py

```diff
@@ -505,10 +508,28 @@ def list_length_validator(cls, v: 'Optional[List[T]]') -> 'Optional[List[T]]':
 
         return v
 
+    @classmethod
+    def unique_items_validator(cls, v: 'Optional[List[T]]') -> 'Optional[List[T]]':
+        try:
```

Comment:
I'm not sure about this "try with a set, then fallback approach", I think it might be cleaner to always use the approach below.

Reply:
alternatively, we could give the user a choice of either they want to use a set or not.

Reply:
Reviewed below.